head/src/main/kotlin/io/qalipsis/core/head/campaign/scheduler/CampaignScheduler.kt

```diff
@@ -24,12 +24,28 @@ internal interface CampaignScheduler {
      * Schedule a campaign test to execute frequently.
      *
      * @param tenant tenant owning the campaign to schedule
-     * @param configurer consider the user's name who configure the campaign
+     * @param configurer name of the user who configured the campaign
      * @param configuration configuration of the campaign to schedule
      */
     suspend fun schedule(
         tenant: String,
         configurer: String,
         configuration: CampaignConfiguration
     ): RunningCampaign
+
+    /**
+     * Updates a scheduled campaign test.
+     *
+     * @param tenant tenant owning the campaign to schedule
+     * @param configurer name of the user who configured the campaign
+     * @param campaignKey unique identifier of the campaign to be updated
+     * @param configuration configuration of the campaign schedule to be updated
+     */
+    suspend fun update(
+        tenant: String,
+        configurer: String,
+        campaignKey: CampaignKey,
+        configuration: CampaignConfiguration
+    ): RunningCampaign
+
```

Comment:
Why not setting the cancel function also here instead of having it in the preparator?

Reply:
That was the first implementation and I ran into the circular dependency issue.

Reply:
I get it, but it makes the workflow much harder to understand, and there might be a way to fix it while keeping the code consistent.
At least, I would not keep the  campaignKeyToJobMap in the preparator but create a class called "ScheduledCampaignsRegistry" that only handles the campaigns and their jobs to caching and listing.